gateway-js/src/__tests__/executeQueryPlan.test.ts

```diff
@@ -142,6 +142,73 @@ describe('executeQueryPlan', () => {
       expect(response).toHaveProperty('errors.0.extensions.variables', {});
     });
 
+    it(`should not send request to downstream services when all entities are undefined`, async () => {
+      overrideResolversInService('reviews', {
+        User: {
+          reviews() {//We are using an example where a past product was removed from the system
+            return [{id:1,body: 'Love it!', product: undefined}, {id:1,body: 'Love it!', product: { __typename: 'Furniture', upc: '1'}}];
+          },
+        },
+      });
```

Comment:
If you wanted to try and check the number of times this resolver is invoked (which I believe would be representative of ensuring there weren't unexpected `FetchNode` invocations), I think you could [use a `jest.fn(...)` Jest Mock function](https://jestjs.io/docs/mock-functions) with the resolver implementation passed to it so you can watch its calls.

```suggestion
	  const reviews = jest.fn(() => ([
	    { id: 1, body: 'Love it!', product: undefined },
	    { id: 1, body: 'Love it!', product: { __typename: 'Furniture', upc: '1'} }
	  ]);
	  
      overrideResolversInService('reviews', {
        User: {
          reviews,
        },
      });
```

... further down, you can:

```js
      expect(reviews.mock.calls.length).toBe(/* The expected number here */);
```